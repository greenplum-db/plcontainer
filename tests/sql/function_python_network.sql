\! plcontainer configure -f $(pwd)/plcontainer_configuration_network.xml -y

drop table if exists test_python_network;

create table test_python_network(i int);
insert into test_python_network select generate_series(0,10);

create or replace function python_network_test1(i integer) returns integer as $$
#container: plc_python_network
return i + 1024
$$ language plcontainer;

create or replace function python_network_test2() returns integer as $$
#container: plc_python_network
return 1024
$$ language plcontainer;

select python_network_test1(i) from test_python_network;
select python_network_test2() from test_python_network;

-- Test permission.
CREATE OR REPLACE FUNCTION py_shared_path_perm_network() RETURNS integer AS $$
# container: plc_python_network
import os
os.open("/tmp/plcontainer/test_file", os.O_RDWR|os.O_CREAT)
return 0
$$ LANGUAGE plcontainer;

select py_shared_path_perm_network();

\! plcontainer configure --restore -y
