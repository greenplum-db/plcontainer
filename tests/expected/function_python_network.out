\! plcontainer configure -f $(pwd)/plcontainer_configuration_test.xml -y
drop table if exists test_python_network;
NOTICE:  table "test_python_network" does not exist, skipping
create table test_python_network(i int);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'i' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
insert into test_python_network select generate_series(0,10);
create or replace function python_network_test1(i integer) returns integer as $$
#container: plc_python_network
return i + 1024
$$ language plcontainer;
create or replace function python_network_test2() returns integer as $$
#container: plc_python_network
return 1024
$$ language plcontainer;
select python_network_test1(i) from test_python_network;
 python_network_test1 
----------------------
                 1024
                 1025
                 1026
                 1027
                 1028
                 1029
                 1030
                 1031
                 1032
                 1033
                 1034
(11 rows)

select python_network_test2() from test_python_network;
 python_network_test2 
----------------------
                 1024
                 1024
                 1024
                 1024
                 1024
                 1024
                 1024
                 1024
                 1024
                 1024
                 1024
(11 rows)

-- Test permission.
CREATE OR REPLACE FUNCTION py_shared_path_perm_network() RETURNS integer AS $$
# container: plc_python_network
import os
os.open("/tmp/plcontainer/test_file", os.O_RDWR|os.O_CREAT)
return 0
$$ LANGUAGE plcontainer;
select py_shared_path_perm_network();
ERROR:  PL/Container client exception occurred:
DETAIL:  
 Exception occurred in Python during function execution 
 Traceback (most recent call last):
  File "<string>", line 5, in py_shared_path_perm_network
OSError: [Errno 2] No such file or directory: '/tmp/plcontainer/test_file'
\! plcontainer configure --restore -y
